src/synapse.py

```diff
@@ -83,6 +91,46 @@ def execute_migrate_config(self, container: ops.Container) -> None:
                 "Migrate config failed, please review your charm configuration"
             )
 
+    def server_name_configured(self, container: ops.Container) -> str | None:
+        """Get server_name from configuration file.
+
+        Args:
+            container: Container of the charm.
+
+        Returns:
+            str | None: server_name or None if configuration file is not found.
+        """
+        try:
+            configuration_content = str(
+                container.pull(SYNAPSE_CONFIG_PATH, encoding="utf-8").read()
+            )
+        except PathError:
+            logger.error("configuration file %s does not exist", SYNAPSE_CONFIG_PATH)
+            return None
+        return yaml.safe_load(configuration_content)["server_name"]
+
+    def reset_instance_action(self, container: ops.Container) -> typing.Dict:
+        """Erase data and config server_name.
+
+        Args:
+            container: Container of the charm.
+
+        Returns:
+            Dictionary containing the execution results for each of the operations executed.
+        """
+        results = {
+            "reset-instance": False,
+        }
+        logging.debug("Erasing directory %s", SYNAPSE_CONFIG_DIR)
+        try:
+            container.remove_path(SYNAPSE_CONFIG_DIR, recursive=True)
+        except PathError as path_error:
+            logger.debug(
+                "exception while erasing directory %s: %s", SYNAPSE_CONFIG_DIR, path_error
+            )
+        results["reset-instance"] = True
```

Comment:
Isn't this supposed to be inside the try block? The result would otherwise be always true

Reply:
Thanks, you made me realize that ignoring this error is not a good approach. I changed the action and now the charm will be blocked and the event will fail if something goes wrong while deleting the directory.